parquet-thrift/src/main/java/org/apache/parquet/thrift/ThriftSchemaConvertVisitor.java

```diff
@@ -76,22 +78,37 @@
   private final boolean doProjection;
   private final boolean keepOneOfEachUnion;
 
-  private ThriftSchemaConvertVisitor(FieldProjectionFilter fieldProjectionFilter, boolean doProjection, boolean keepOneOfEachUnion) {
+  private boolean writeThreeLevelList = ParquetWriteProtocol.WRITE_THREE_LEVEL_LISTS_DEFAULT;
```

Comment:
It seems this variable gets a value in the constructor anyway. I would suggest having it final and assign value in the constructor only once.

Reply:
Good point, made the change.